File: incubating/client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/extension/ProtonClientOptions.kt
package me.proton.drive.sdk.extension

import me.proton.drive.sdk.entity.ProtonClientOptions
import proton.sdk.protonClientOptions
import proton.sdk.telemetry

internal fun ProtonClientOptions.toProtobuf(
    recordMetricAction: Long? = null,
) = protonClientOptions {
    this@toProtobuf.userAgent?.let { userAgent = it }
    this@toProtobuf.baseUrl?.let { baseUrl = it }
    this@toProtobuf.bindingsLanguage?.let { bindingsLanguage = it }
    this@toProtobuf.tlsPolicy?.let { tlsPolicy = it.toProtobuf() }
    telemetry = telemetry {
        this@toProtobuf.loggerProvider?.let { loggerProviderHandle = it.handle }
        recordMetricAction?.let { this@telemetry.recordMetricAction = recordMetricAction }
    }
    this@toProtobuf.entityCachePath?.let { entityCachePath = it }
}

File: client/cs/sdk/src/Proton.Sdk/ProtonClientOptions.cs
using Proton.Sdk.Caching;
using Proton.Sdk.Http;
using Proton.Sdk.Telemetry;

namespace Proton.Sdk;

public record ProtonClientOptions
{
    public Uri? BaseUrl { get; set; }
    public string? UserAgent { get; set; }
    public ProtonClientTlsPolicy? TlsPolicy { get; set; }
    public Func<DelegatingHandler>? CustomHttpMessageHandlerFactory { get; set; }
    public IHttpClientFactory? HttpClientFactory { get; set; }
    public ICacheRepository? EntityCacheRepository { get; set; }
    public ITelemetry? Telemetry { get; set; }
    public IFeatureFlagProvider? FeatureFlagProvider { get; set; }
    public string? BindingsLanguage { get; set; }
}
